symfit/core/fit_results.py

```diff
@@ -17,32 +22,43 @@ class FitResults(object):
     their optimized values. Can be `**` unpacked when evaluating
     :class:`~symfit.core.fit.Model`'s.
     """
-    def __init__(self, model, popt, covariance_matrix, infodict, mesg, ier, minimizer, objective, **gof_qualifiers):
+
+    def __init__(self, model, popt, covariance_matrix, minimizer_output, minimizer, objective):
         """
         :param model: :class:`~symfit.core.fit.Model` that was fit to.
         :param popt: best fit parameters, same ordering as in model.params.
-        :param pcov: covariance matrix.
-        :param infodict: dict with fitting info.
-        :param mesg: Status message.
-        :param ier: Number of iterations.
+        :param covariance_matrix: covariance matrix.
+        :param minimizer_output: Raw output as given by the minimizer. This is
+            assumed to be a :class:`~scipy.optimize.OptimizationResult` instance.
```

Comment:
Now symfit becomes Scipy specific, barring the use of any other minimizer library. This seems undesirable.

Reply:
Good point, that is definitely not the intent. Internally we do assume this object for now, and in the near feature we pick attributes from `minimizer_output` based on the type of `minimizer`.

Reply:
How about feeding minimizer output as keywords to FitResult? That way the unpacking from minimizer output becomes the responsibility of Minimizer objects, and it will still be compatible with frameworks that produce dicts.

Reply:
You mean that `minimizer_output` is a dict, or that all minimizer output should be fed to FitResults as keywords? I noticed that the latter was becoming a mess, which is why I switched to this, but making minimizer_output into a standardized dict might be nice. This is also in fact what they finally have decided to do at scipy, because the more recent version of `leastsq` now also returns a `OptimizeResult`, which is just a dict with fancy printing. However, we don't get that yet because we use the `leastsqbound` package here. Switching to scipy's current support for bounds is another future PR.

Reply:
> You mean that minimizer_output is a dict, or that all minimizer output should be fed to FitResults as keywords? I noticed that the latter was becoming a mess, which is why I switched to this, but making minimizer_output into a standardized dict might be nice.

I mean the latter. It might make for a very long signature, but it makes documenting what's expected, as well as default arguments easier. The options are functionally equivalent though.

Reply:
The problem I see with that is that even now not every minimizer returns the same stuff, so then we would have to add a lot of arguments to the signature which are usually not provided. Plus I think it's good to have access to the raw output just in case, because we might not always be up to date with changes made to `scipy` or other libraries in the future. Therefore, I think just accepting a dict with all the output is a good way to get around that.

Reply:
Why not both? :)
We can give all arguments used by FitResult a name, and accept everything else as `**kwargs`. That way you can always construct a FitResult with your **minimizer_output, and the API of FitResult becomes fairly clean in what should be provided at minimum. An helper method to rename (some) dict keys might be valuable, but that's for later.

Reply:
Good point, I agree that we should keep mandetory arguments separate, and now that `gof_qualifiers` are calculated by FitResults itself, the kwargs could be the remaining minimizer output.